### PR TITLE
Allow to indent and align with tabs instead of spaces

### PR DESCRIPTION
**Pull request by Vampire**

### CODE REVIEW

File: javaparser-core/src/main/java/com/github/javaparser/printer/PrettyPrinterConfiguration.java [hunk 1]
```diff
@@ -82,6 +126,39 @@ public PrettyPrinterConfiguration setIndentSize(int size) {
...
+     */
+    public int getTabWidth() {
+        return tabWidth;
+    }
+
+    /**
+     * Set the tab width for pretty aligning.
+     */
+    public PrettyPrinterConfiguration setTabWidth(int tabWidth) {
+        if (tabWidth <= 0) {
```
Comment (matozoid) [inline javaparser-core/src/main/java/com/github/javaparser/printer/PrettyPrinterConfiguration.java]: Yep, time to create that `assertPositive` you mentioned.

File: javaparser-core/src/main/java/com/github/javaparser/printer/PrettyPrinterConfiguration.java [hunk 2]
```diff
@@ -47,12 +69,20 @@
     /**
      * Set the string to use for indenting. For example: "\t", "    ", "".
      *
-     * @deprecated use setIndentSize. Tabs are no longer supported.
+     * @deprecated use {@link #setIndentSize(int)}.
```
Comment (matozoid) [inline javaparser-core/src/main/java/com/github/javaparser/printer/PrettyPrinterConfiguration.java]: and `setIndentType`, I guess

File: javaparser-core/src/main/java/com/github/javaparser/printer/PrettyPrinterConfiguration.java (hunk 1 above)
Comment (Vampire) [reply javaparser-core/src/main/java/com/github/javaparser/printer/PrettyPrinterConfiguration.java]: Well, it was `assertNonNegative` for `indentSize` and is `assertPositive` for `tabWidth`.
Actually I had added `assertPositive` to `Util`, but changed it to your style.
Want me to add it back in?

File: javaparser-core/src/main/java/com/github/javaparser/printer/PrettyPrinterConfiguration.java [hunk 3]
```diff
@@ -25,20 +25,42 @@
...
+        SPACES,
+
+        /**
+         * Indent with tabs as far as possible.
+         * For proper aligning, the tab width is necessary and by default 4.
+         */
+        TABS,
+
+        /**
+         * Indent with tabs but align with spaces.
```
Comment (matozoid) [inline javaparser-core/src/main/java/com/github/javaparser/printer/PrettyPrinterConfiguration.java]: I'm not sure some random user will understand the difference between indenting and aligning, and how this setting will work out.

**Comment by matozoid** [comment]
Cool!

File: javaparser-core/src/main/java/com/github/javaparser/printer/PrettyPrinterConfiguration.java (hunk 3 above)
Comment (Vampire) [reply javaparser-core/src/main/java/com/github/javaparser/printer/PrettyPrinterConfiguration.java]: Well, random user has a hard time to get started anyway. :-D

How do you like
```java
        /**
         * Indent with tabs but align with spaces when wrapping and aligning
         * method call chains and method call parameters.
         * 
         * <p/><i>Example result:</i>
         * <pre>
         * class Foo {
         * 
         * \tvoid bar() {
         * \t\tfoo().bar()
         * \t\t......baz(() -> {
         * \t\t..........\tboo().baa()
         * \t\t..........\t......bee(a,
         * \t\t..........\t..........b,
         * \t\t..........\t..........c);
         * \t\t..........})
         * \t\t......bam();
         * \t}
         * }
         * </pre>
         */
```
?

Comment (matozoid) [reply javaparser-core/src/main/java/com/github/javaparser/printer/PrettyPrinterConfiguration.java]: Are you sure anybody wants that setting? ;-) But yeah, that's fine.

File: javaparser-core/src/main/java/com/github/javaparser/printer/PrettyPrinterConfiguration.java (hunk 1 above)
Comment (matozoid) [reply javaparser-core/src/main/java/com/github/javaparser/printer/PrettyPrinterConfiguration.java]: The codebase is a mix of styles from 84 contributors, don't worry :-) Yes please!

File: javaparser-core/src/main/java/com/github/javaparser/printer/PrettyPrinterConfiguration.java (hunk 3 above)
Comment (Vampire) [reply javaparser-core/src/main/java/com/github/javaparser/printer/PrettyPrinterConfiguration.java]: I think so, yes.
It was like that basically before when tabs were configured.
Well, only the inital indent was in tabs and remaining stuff in spaces completely afair.
Now you have tabs for indentation and any viewer can choose the tab size he likes, while still having the call chains and parameters aligned. :-)

**Comment by matozoid** [comment]
Thanks!

**Pull request merged**